tests/unit_tests/test_connection.py

```diff
@@ -1,31 +1,37 @@
-from dataregistry.db_basic import DbConnection
 import os
 
+import pytest
+from dataregistry.db_basic import DbConnection
 
-def test_connect_manual():
-    """Test connection to database with manually passed config location"""
-    conn = DbConnection(
-        config_file=os.path.join(os.getenv("HOME"), ".config_reg_access"), verbose=True
-    )
-    assert conn.engine is not None
-    assert conn.dialect is not None
-    if conn.dialect != "sqlite":
-        assert conn.schema is not None
+# This is always assumed to exist from the one-time-setup for the dataregistry
+_DEFAULT_LOCATION = os.path.join(os.getenv("HOME"), ".config_reg_access")
 
 
-def test_connect_env():
-    """Test connection to database with DATAREG_CONFIG env set"""
-    os.environ["DATAREG_CONFIG"] = os.path.join(os.getenv("HOME"), ".config_reg_access")
-    conn = DbConnection(verbose=True)
-    assert conn.engine is not None
-    assert conn.dialect is not None
-    if conn.dialect != "sqlite":
-        assert conn.schema is not None
+@pytest.mark.parametrize(
+    "config_file,set_env_var",
+    [
+        (_DEFAULT_LOCATION, False),
+        (None, False),
+        (None, True),
+    ],
+)
+def test_connection(config_file, set_env_var):
+    """
+    Test various ways of passing the connection config file
```

Comment:
Since all three cases boil down to the same thing - using the file at ${HOME}/.config_reg_access - there is no way to tell if the code in DbConnection is really following a different path other than looking at the output from the verbose setting.  How about creating a file (could have the same contents) in a different location for each case and eliminating other two, so for each call either it finds the file in the specified location or fails?